components/NavLayout/SideBar.js

```diff
@@ -1,12 +1,122 @@
+
+  // hard coded data, eventually change sampleUser to [username]
+  const pageUrls = {
+    0: "/sampleUser/my-classes",
+    1: "/sampleUser/calendar",
+    2: "/sampleUser/notifications",
+    3: "/sampleUser/class-catalog"
+  }
+
+  const sideBarIcons = {
+    0: <ClassOutlinedIcon style={{ color: '#FFFFFF' }} />,
+    1: <CalendarMonthOutlinedIcon style={{ color: '#FFFFFF' }} />,
+    2: <NotificationsOutlinedIcon style={{ color: '#FFFFFF' }} />,
+    3: <SearchOutlinedIcon style={{ color: '#FFFFFF' }} />
+  }
+
+  // TODO: change "Current User" to [username]'s name
+  // TODO: change avatar src to dynamically render user's profile picture ? defaultProfilePic
   return (
-    <nav className={styles.nav}>
-      <h3>My Classes</h3>
-      <h3>Calendar</h3>
-      <h3>Notifications</h3>
-      <h3>Class Catalog</h3>
-    </nav>
+    <Box sx={{ display: 'flex' }}>
+      <CssBaseline />
+      <AppBar position="fixed" open={open} className={styles.topbar}>
+        <Toolbar>
+          <IconButton
+            color="inherit"
+            aria-label="open drawer"
+            onClick={handleDrawerOpen}
+            edge="start"
+            sx={{
+              marginRight: 5,
+              ...(open && { display: 'none' }),
+            }}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography className={styles.siteTitle} variant="h5" noWrap component="div">
+            ClassMates
+          </Typography>
+          <Container className={styles.profileIcon} sx={{ flexGrow: 0 }}>
+            <Typography className={styles.profileName}>Current User</Typography>
+            <Avatar
+              className={styles.profileAvatar}
+              alt="Remy Sharp"
+              src={defaultProfilePic}
+              sx={{ width: 50, height: 50 }}
+            />
+          </Container>
+        </Toolbar>
+      </AppBar>
+      <Drawer className='drawer' variant="permanent" open={open} >
+        <DrawerHeader>
+          <IconButton onClick={handleDrawerClose}>
+            {theme.direction === 'rtl' ? <ChevronRightIcon style={{ color: '#FFFFFF' }} /> : <ChevronLeftIcon style={{ color: '#FFFFFF' }} />}
+          </IconButton>
+        </DrawerHeader>
+
+        <List>
+          {['My Classes', 'Calendar', 'Notifications', 'Class Catalog'].map((text, index) => (
+            <ListItemButton
+              key={text}
+              sx={{
+                minHeight: 48,
+                justifyContent: open ? 'initial' : 'center',
+                px: 2.5,
+              }}
+            >
+              <ListItemIcon
+                sx={{
+                  minWidth: 0,
+                  mr: open ? 3 : 'auto',
+                  justifyContent: 'center',
+                }}
+              >
+                <Link href={pageUrls[index]}>{sideBarIcons[index]}</Link>
+              </ListItemIcon>
+              <Link href={pageUrls[index]}><ListItemText primary={text} sx={{ opacity: open ? 1 : 0 }} /></Link>
```

Comment:
Could we wrap the <Link /> around the text and icon? If you can't for some reason, drop a comment in the code explaining why.

Reply:
Yeah! we tried that in the beginning, but the code breaks when everything is included in one line, so the separation is intentional